prime-router/src/main/kotlin/metadata/LivdLookup.kt

```diff
@@ -0,0 +1,206 @@
+package gov.cdc.prime.router.metadata
+
+import gov.cdc.prime.router.Metadata
+
+// this will be used by the LivdMappers and the CustomFHIRFunction
+// /need to move most of LivdMappers in here
+// will need to move the tests to look at this as well
+// keep the library the way it was then use the context to use the library to pass in custom functions
+object LivdLookup {
+    fun find(
+        testPerformedCode: String?,
+        processingModeCode: String?,
+        deviceId: String?,
+        equipmentModelId: String?,
+        testKitNameId: String?,
+        equipmentModelName: String?,
+        tableColumn: String,
+        tableRef: LookupTable? = Metadata.getInstance().findLookupTable(name = "LIVD-SARS-CoV-2")
+    ): String? {
+        if (tableRef == null) {
+            return null
+        }
+
+        val filters = tableRef.FilterBuilder()
+        // get the test performed code for additional filtering of the test information in case we are
+        // dealing with tests that check for more than one type of disease, for example COVID + influenza
+        if (testPerformedCode != null) {
+            filters.equalsIgnoreCase(LivdTableColumns.TEST_PERFORMED_CODE.colName, testPerformedCode)
+        }
+
+        // If the data is NOT flagged as test data then ignore any test devices in the LIVD table
+        if (processingModeCode == null ||
+            processingModeCode.uppercase() != testProcessingModeCode
+        ) {
+            filters.notEqualsIgnoreCase(
+                LivdTableColumns.PROCESSING_MODE_CODE.colName,
+                testProcessingModeCode
+            )
+        }
+
+        // carry on as usual
+        val filtersCopy = filters.copy() // Filters are not reusable
+        return when {
+            !deviceId.isNullOrEmpty() -> lookupByDeviceId(
+                tableColumn,
+                deviceId,
+                filtersCopy
+            )
+            !equipmentModelId.isNullOrEmpty() -> lookupByEquipmentUid(
+                tableColumn,
+                equipmentModelId,
+                filtersCopy
+            )
+            !testKitNameId.isNullOrEmpty() -> lookupByTestkitId(
+                tableColumn,
+                testKitNameId,
+                filtersCopy
+            )
+            !equipmentModelName.isNullOrEmpty() -> lookupByEquipmentModelName(
+                tableColumn, equipmentModelName, filtersCopy
+            )
+
+            else -> null
+        }
+    }
+
+    private val standard99ELRTypes = listOf("EUA", "DII", "DIT", "DIM", "MNT", "MNI", "MNM")
+
+    /**
+     * The value for the processing mode code for test data.
+     */
+    internal const val testProcessingModeCode = "T"
+
+    /**
+     * Does a lookup in the LIVD table based on the element Id
+     * @param element the schema element to use for lookups
```

Comment:
I think element has been removed as a param for all of the lookup methods. Can you update all of the corresponding kdocs?